### PR TITLE
isort

### PR DESCRIPTION
**Pull request by maxim-lian**
Not sure if we want this?

Probably too strict to enforce on every commit, more permissible to do a one-time update, assuming it doesn't cause merge conflicts. 

You can get the same result by running `isort -y`

### CODE REVIEW

**Comment by maxim-lian** [comment]
Requires https://github.com/pydata/xarray/pull/1925 to make @stickler-ci happy (and this needs to stop the `I002` disabling when that is merged)

**Comment by maxim-lian** [comment]
Updated

**Comment by shoyer** [comment]
Is it possible to enforce this programmatically, e.g., with flake8?

I'm a little surprised flake8 doesn't already require sorted imports by default. We use a modified version of pylint internally at Google that does this.

File: xarray/tests/test_coding_times.py [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 import pandas as pd
+import pytest
 
 from xarray import Variable, coding
 from xarray.coding.times import _import_netcdftime
+
 from . import (
-    TestCase, requires_netcdftime, assert_array_equal)
-import pytest
+    TestCase, assert_array_equal, requires_netCDF4, requires_netcdftime)
```
Comment (jhamman) [inline xarray/tests/test_coding_times.py]: if you remove `requires_netCDF4`, I think the flake8 tests will pass.

**Comment by dopplershift** [comment]
I recommend [flake8-import-order](https://github.com/PyCQA/flake8-import-order). If you install that plugin, then flake8 will enforce import ordering and grouping.

**Comment by maxim-lian** [comment]
We use `flake8-isort`, which implements `isort` itself in `flake8`. @dopplershift do you have a view between them?

One advantage of `isort` is that there's a link between the formatter and the checker, so running `isort` can never cause errors.

But stepping back, we should think about whether we want to mandate this on every PR. It doesn't matter _that_ much (less than flake8 imo), and it can be frustrating to have formatting errors get in the way of a PR. A tool that automatically corrects the code like @stickler-ci is great, but it doesn't support this one. 

A compromise could be to put it in allowed failures in Travis? Or if people think we should add this, let's do it.

**Comment by dopplershift** [comment]
I don't have much preference between the two--I picked one and it worked for me. I just fixed up the initial import cleanups by hand and didn't find the need for a tool to do so.

In my experience, having formatting errors hold up a PR hasn't been a problem. IMO, import ordering is a more important style issue than many of the things that flake8 catches.

**Comment by shoyer** [comment]
I would also be OK with just fixing this up in occasional cleanup PR.

If Stickler-CI supported this that might be a decisive consideration in terms of enforcing it. Certainly it's a better experience for contributors to get formatting errors as comments on a PR rather than a failed build.

**Comment by maxim-lian** [comment]
> a better experience for contributors to get formatting errors as comments on a PR

+ the really cool feature is that it'll add commits with changes, where it can

**Comment by maxim-lian** [comment]
> I would also be OK with just fixing this up in occasional cleanup PR.

Would you prefer in allowed failures, or that's a bad middle-ground?

**Comment by maxim-lian** [comment]
@jhamman @shoyer green

**Comment by maxim-lian** [comment]
@jhamman @shoyer Green!

No auto-check

**Pull request merged**